ejb/WFLY-6892-ejb-access-logging.adoc

```diff
@@ -0,0 +1,138 @@
+= EJB Access Logging
+:author:            Cheng Fang
+:email:             cfang@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-6892 (Access logging for EJBs)
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFCORE-4355 (Add utility for event logging such as audit and access logging)
+* https://issues.jboss.org/browse/EAP7-523 (Access logging for EJBs)
+* https://issues.jboss.org/browse/EAP7-699 (Enhanced Audit Logging - RFC support and reliability/speed customization)
+* https://issues.jboss.org/browse/WFLY-11031 (Json encoding of Access Log Events)
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:szhantem@redhat.com[Sultan Zhantemirov]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[x] QE
+
+=== Affected Projects or Components
+
+* ejb3
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* implement the access logging feature similar to WildFly web access logging.
+
+* add access-logging definition to wildfly-ejb3 schema (see next section below)
+
+* configure ejb access logging with CLI
+
+* enable and disable ejb access logging
+
+* configure how to rotate ejb access logs if applicable
+
+** default should be OFF for performance consideration
+
+* configure output location and format
+
+** directory/file
+
+** stdout
+
+* choose from pre-defined logging formats (need to adapt for ejb invocation):
+
+** common (default): %h %l %u %t "%r" %s %b
+
+** combined: %h %l %u %t "%r" %s %b "%{i,Referer}" "%{i,User-Agent}"
+
+* for more customization, define which attributes to include in ejb access logging
+
+** consider adapting the following attributes that are currently supported by web access logging:
```

Comment:
IMO before a pattern like this is defined we need to know what users expect to see. The patterns below to me are not intuitive. These were used in Undertow for legacy reasons AIUI.

That said if we feel we should keep standard patterns that's okay too. We just don't want to use a legacy format because that's how it's been done :)

Reply:
The intent was to stick to convention already in use in undertow and apache web server, which should make it easier for users.  The logging pattern will not be intuitive anyway and so I figured existing convention will be a bit easier.  But I'm open to other better patterns.

Reply:
I think we should only do this if it makes sense. I'm not sure on what details are expected to be logged. If they match what a web container would log then it likely makes sense. If not then we should rethink it IMO.

Reply:
I've updated the proposal to use more readable logging patterns, since we will need to support JSON, which will require a somewhat readable key anyway.  So for a pattern = "date time ip ejb method", a log entry with simple pattern will be like "2019-05-01 20:50:54,975 127.0.0.1 hello hello", a JSON entry will be like: {"date": "2019-05-01", "time": "20:50:54,975", "ip": "127.0.0.1", "ejb": "hello", "method": "hello"}

Reply:
I've updated the proposal to use more readable logging patterns, since we will need to support JSON, which will require a somewhat readable key anyway.  So for a pattern = "date time ip ejb method", a log entry with simple pattern will be like "2019-05-01 20:50:54,975 127.0.0.1 hello hello", a JSON entry will be like: {"date": "2019-05-01", "time": "20:50:54,975", "ip": "127.0.0.1", "ejb": "hello", "method": "hello"}